lib/fastlane/plugin/wpmreleasetoolkit/actions/common/prototype_build_details_comment_action.rb

```diff
@@ -0,0 +1,130 @@
+        end
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Generates the nicely-formatted string providing all the details of a prototype build, ready to be used as a PR comment (e.g. via `comment_on_pr`).'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :appcenter_org_name,
+            env_name: 'APPCENTER_OWNER_NAME', # Same as the one used by the `appcenter_upload` action
+            description: 'The name of the organization in App Center',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :appcenter_app_name,
+            env_name: 'APPCENTER_APP_NAME', # Same as the one used by the `appcenter_upload` action
+            description: 'The name of the app in App Center',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :appcenter_release_id,
+            env_name: 'FL_PROTOTYPE_BUILD_APPCENTER_RELEASE_ID',
+            description: "The release ID/Number in App Center; can be obtained using `lane_context[SharedValues::APPCENTER_BUILD_INFORMATION]['id']`",
+            type: Integer
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :download_url,
+            env_name: 'FL_PROTOTYPE_BUILD_DOWNLOAD_URL',
+            description: 'The URL to download the build directly; e.g. a public link to the `.apk` file; you might use `lane_context[SharedValues::APPCENTER_DOWNLOAD_LINK]` for this for example',
+            type: String,
+            optional: true,
+            default_value: nil
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :fold,
+            env_name: 'FL_PROTOTYPE_BUILD_DETAILS_COMMENT_FOLD',
+            description: 'If true, will wrap the HTML table inside a <details> block (hidden by default)',
+            type: Boolean,
+            default_value: false
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :metadata,
+            env_name: 'FL_PROTOTYPE_BUILD_DETAILS_COMMENT_METADATA',
+            description: 'All additional metadata (as key/value pairs) you want to include in the HTML table of the comment',
+            type: Hash,
+            optional: true,
+            default_value: {}
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :footnote,
+            env_name: 'FL_PROTOTYPE_BUILD_DETAILS_COMMENT_FOOTNOTE',
+            description: 'Optional footnote to add below the HTML table of the comment',
+            type: String,
+            default_value: '<em>Automatticians: You can use our internal self-serve MC tool to give yourself access to App Center if needed.</em>'
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :commit,
+            env_name: 'BUILDKITE_COMMIT',
```

Comment:
Worth pointing out that this adds an implicit dependency on Buildkite as the CI provider.

It's a pretty safe assumption to make, given this is an internal tool only.

Reply:
Tbh I initially thought about writing this as `env_names: ['BUILDKITE_COMMIT', …]` and add support for the env vars used by more CI providers. But I figured that I had other fishes to fry to chase down the env vars used by different CI providers while we wouldn't need it internally.

I'm still gonna add a comment to make that clearer though.

Reply:
Updated with comments in 854098f3e5995f3e45eaa5b893ed8c50e906a2aa